### PR TITLE
Wip rgw create bucket dpp log

### PR DESCRIPTION
**Pull request by TRYTOBE8TME**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/librgw.cc [hunk 1]
```diff
@@ -241,7 +242,7 @@ namespace rgw {
     /* XXX and -then- stash req_state pointers everywhere they are needed */
     ret = req->init(rgw_env, &rados_ctx, io, s);
     if (ret < 0) {
-      dout(10) << "failed to initialize request" << dendl;
+      ldpp_dout(s, 10) << "failed to initialize request" << dendl;
```
Comment (alimaredia) [inline src/rgw/librgw.cc:245]: @cbodley should he be using `op` here instead of `s`?

File: src/rgw/rgw_admin.cc [hunk 2]
```diff
@@ -1373,7 +1373,7 @@ int check_min_obj_stripe_size(rgw::sal::RGWRadosStore *store, RGWBucketInfo& buc
     auto biter = bl.cbegin();
     decode(manifest, biter);
   } catch (buffer::error& err) {
-    ldpp_dout(store->ctx(), 0) << "ERROR: failed to decode manifest" << dendl;
+    ldout(store->ctx(), 0) << "ERROR: failed to decode manifest" << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_admin.cc]: all of these in `rgw_admin.cc` should be using the `dpp()` declared globablly near the top of the file. So these should be `ldpp_dout(dpp(), 0)`

File: src/rgw/rgw_admin.cc [hunk 3]
```diff
@@ -2363,7 +2363,7 @@ static void get_hint_entities(const std::set<rgw_zone_id>& zones, const std::set
       rgw_bucket hint_bucket;
       int ret = init_bucket(b, hint_bucket_info, hint_bucket);
       if (ret < 0) {
-	ldpp_dout(store->ctx(), 20) << "could not init bucket info for hint bucket=" << b << " ... skipping" << dendl;
+	ldout(store->ctx(), 20) << "could not init bucket info for hint bucket=" << b << " ... skipping" << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_admin.cc]: Ditto.

File: src/rgw/rgw_admin.cc [hunk 4]
```diff
@@ -2469,7 +2469,7 @@ static int sync_info(std::optional<rgw_zone_id> opt_target_zone, std::optional<r
     RGWBucketSyncPolicyHandlerRef hint_bucket_handler;
     int r = store->ctl()->bucket->get_sync_policy_handler(zid, hint_bucket, &hint_bucket_handler, null_yield, dpp());
     if (r < 0) {
-      ldpp_dout(store->ctx(), 20) << "could not get bucket sync policy handler for hint bucket=" << hint_bucket << " ... skipping" << dendl;
+      ldout(store->ctx(), 20) << "could not get bucket sync policy handler for hint bucket=" << hint_bucket << " ... skipping" << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_admin.cc:2472]: Ditto.

File: src/rgw/rgw_admin.cc [hunk 5]
```diff
@@ -2765,7 +2765,7 @@ static int scan_totp(CephContext *cct, ceph::real_time& now, rados::cls::otp::ot
                                pins[1].c_str());
       if (rc != OATH_INVALID_OTP) {
         *pofs = time_ofs - step_size + step_size * totp.window / 2;
-        ldpp_dout(cct, 20) << "found at time=" << start_time - time_ofs << " time_ofs=" << time_ofs << dendl;
+        ldout(cct, 20) << "found at time=" << start_time - time_ofs << " time_ofs=" << time_ofs << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_admin.cc]: Ditto.

File: src/rgw/rgw_admin.cc [hunk 6]
```diff
@@ -6864,7 +6864,7 @@ int main(int argc, const char **argv)
           if (min_rewrite_stripe_size > 0) {
             r = check_min_obj_stripe_size(store, bucket_info, &obj, min_rewrite_stripe_size, &need_rewrite);
             if (r < 0) {
-              ldpp_dout(store->ctx(), 0) << "WARNING: check_min_obj_stripe_size failed, r=" << r << dendl;
+              ldout(store->ctx(), 0) << "WARNING: check_min_obj_stripe_size failed, r=" << r << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_admin.cc]: Ditto.

File: src/rgw/rgw_admin.cc [hunk 7]
```diff
@@ -6745,7 +6745,7 @@ int main(int argc, const char **argv)
         return -ret;
       }
     } else {
-      ldpp_dout(store->ctx(), 20) << "skipped object" << dendl;
+      ldout(store->ctx(), 20) << "skipped object" << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_admin.cc]: Ditto.

File: src/rgw/rgw_admin.cc [hunk 8]
```diff
@@ -6735,7 +6735,7 @@ int main(int argc, const char **argv)
     if (min_rewrite_stripe_size > 0) {
       ret = check_min_obj_stripe_size(store, bucket_info, &obj, min_rewrite_stripe_size, &need_rewrite);
       if (ret < 0) {
-        ldpp_dout(store->ctx(), 0) << "WARNING: check_min_obj_stripe_size failed, r=" << ret << dendl;
+        ldout(store->ctx(), 0) << "WARNING: check_min_obj_stripe_size failed, r=" << ret << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_admin.cc]: Ditto.

File: src/rgw/rgw_rest_user.cc [hunk 9]
```diff
@@ -981,7 +981,7 @@ void RGWOp_Quota_Set::execute(optional_yield y)
   RGWUser user;
   op_ret = user.init(s, store, op_state, y);
   if (op_ret < 0) {
-    ldout(store->ctx(), 20) << "failed initializing user info: " << op_ret << dendl;
+    ldpp_dout(s, 20) << "failed initializing user info: " << op_ret << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_rest_user.cc:984]: you might want to use `this` here instead of `s`.

File: src/rgw/rgw_rest_user.cc [hunk 10]
```diff
@@ -995,7 +995,7 @@ void RGWOp_Quota_Set::execute(optional_yield y)
     UserQuotas quotas;
 
     if ((op_ret = rgw_rest_get_json_input(store->ctx(), s, quotas, QUOTA_INPUT_MAX_LEN, NULL)) < 0) {
-      ldout(store->ctx(), 20) << "failed to retrieve input" << dendl;
+      ldpp_dout(s, 20) << "failed to retrieve input" << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_rest_user.cc:998]: ditto

File: src/rgw/rgw_rest_user.cc [hunk 11]
```diff
@@ -1008,7 +1008,7 @@ void RGWOp_Quota_Set::execute(optional_yield y)
       bool empty;
       op_ret = rgw_rest_get_json_input(store->ctx(), s, quota, QUOTA_INPUT_MAX_LEN, &empty);
       if (op_ret < 0) {
-        ldout(store->ctx(), 20) << "failed to retrieve input" << dendl;
+        ldpp_dout(s, 20) << "failed to retrieve input" << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_rest_user.cc:1011]: ditto

File: src/rgw/rgw_rest_user.cc [hunk 12]
```diff
@@ -1022,7 +1022,7 @@ void RGWOp_Quota_Set::execute(optional_yield y)
       string err_msg;
       op_ret = user.info(info, &err_msg);
       if (op_ret < 0) {
-        ldout(store->ctx(), 20) << "failed to get user info: " << op_ret << dendl;
+        ldpp_dout(s, 20) << "failed to get user info: " << op_ret << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_rest_user.cc:1025]: ditto

File: src/rgw/rgw_rest_user.cc [hunk 13]
```diff
@@ -1053,7 +1053,7 @@ void RGWOp_Quota_Set::execute(optional_yield y)
   string err;
   op_ret = user.modify(s, op_state, y, &err);
   if (op_ret < 0) {
-    ldout(store->ctx(), 20) << "failed updating user info: " << op_ret << ": " << err << dendl;
+    ldpp_dout(s, 20) << "failed updating user info: " << op_ret << ": " << err << dendl;
```
Comment (alimaredia) [inline src/rgw/rgw_rest_user.cc:1056]: ditto

File: src/rgw/rgw_orphan.cc [hunk 14]
```diff
@@ -564,16 +564,16 @@ int RGWOrphanSearch::build_linked_oids_for_bucket(const DoutPrefixProvider *dpp,
...
+        ldpp_dout(dpp, 20) << "obj entry: " << entry.key.name << " [" << entry.key.instance << "]" << dendl;
       }
 
-      ldout(store->ctx(), 20) << __func__ << ": entry.key.name=" << entry.key.name << " entry.key.instance=" << entry.key.instance << dendl;
+      ldpp_dout(dpp, 20) << __func__ << ": entry.key.name=" << entry.key.name << " entry.key.instance=" << entry.key.instance << dendl;
 
       if (!detailed_mode &&
           entry.meta.accounted_size <= (uint64_t)store->ctx()->_conf->rgw_max_chunk_size) {
-        ldout(store->ctx(),5) << __func__ << "skipping stat as the object " << entry.key.name
+        ldpp_dout(this, 5) << __func__ << "skipping stat as the object " << entry.key.name
```
Comment (alimaredia) [inline src/rgw/rgw_orphan.cc]: why does `this` work here? I don't see `RGWOrphanSeach inheriting fomr `RGWOp`. I'm not sure how this compiles am I missing something?

File: src/rgw/rgw_file.cc [hunk 15]
```diff
@@ -1942,7 +1942,7 @@ namespace rgw {
       cs_info.blocks = std::move(compressor->get_compression_blocks());
       encode(cs_info, tmp);
       attrs[RGW_ATTR_COMPRESSION] = tmp;
-      ldout(state->cct, 20) << "storing " << RGW_ATTR_COMPRESSION
+      ldpp_dout(this, 20) << "storing " << RGW_ATTR_COMPRESSION
```
Comment (alimaredia) [inline src/rgw/rgw_file.cc:1945]: @cbodley `RGWWriteRequest` is a `RGWPutObj`, should we remove the dpp in this function call or leave it?

**Comment by alimaredia** [comment]
@TRYTOBE8TME A few things besides the comments:
- I'm not sure why my comments in `rgw_admin.cc` became outdated but they still stand. 
- I also don't know why your PR includes my two commit could you remove them
- There's merge conflicts on this PR, make sure to resolve them, but do that after removing my commits.

**Review by alimaredia — CHANGES_REQUESTED** [review]
@TRYTOBE8TME A few things besides the comments:
    - I'm not sure why my comments in rgw_admin.cc became outdated but they still stand.
    - I also don't know why your PR includes my two commit could you remove them
    - There's merge conflicts on this PR, make sure to resolve them, but do that after removing my commits
    - the 17th commit with changes to rgw_realm_reloader.cc is empty. I'm not sure why this is.

File: src/rgw/rgw_admin.cc (hunk 2 above)
Comment (TRYTOBE8TME) [reply src/rgw/rgw_admin.cc]: Okay

File: src/rgw/rgw_orphan.cc (hunk 14 above)
Comment (TRYTOBE8TME) [reply src/rgw/rgw_orphan.cc]: This is outdated, it is `dpp` in the main file.

File: src/rgw/rgw_rest_user.cc (hunk 9 above)
Comment (TRYTOBE8TME) [reply src/rgw/rgw_rest_user.cc:984]: Okay.

**Comment by alimaredia** [comment]
@TRYTOBE8TME you missed quite a few ldpp_dout's in `get_obj_state_impl` in `rgw_rados.cc`. For example grep for "get_obj_state:" in the file. I'd recommend you take a look at every single one of the `ldout`s and `douts` and `lderr`s in `rgw_rados.cc` and see if they can be changed.

**Comment by TRYTOBE8TME** [comment]
New PR: https://github.com/ceph/ceph/pull/39204

File: src/rgw/librgw.cc (hunk 1 above)
Comment (cbodley) [reply src/rgw/librgw.cc:245]: yes

File: src/rgw/rgw_file.cc (hunk 15 above)
Comment (cbodley) [reply src/rgw/rgw_file.cc:1945]: remove

**Pull request closed without merging**